components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts

```diff
@@ -132,22 +133,20 @@ export class ApiTokenListComponent implements OnInit, OnDestroy {
   public createToken(): void {
     this.creatingToken = true;
     const tok = {
-      id: this.createTokenForm.controls['id'].value,
-      name: this.createTokenForm.controls['name'].value.trim(),
+      id: this.createTokenForm.controls.id.value,
+      name: this.createTokenForm.controls.name.value.trim(),
       projects: this.createTokenForm.controls.projects.value
     };
     this.store.dispatch(new CreateToken(tok));
+  }
 
-    const member = stringToMember(`token:${tok.id}`);
-    const policies: string[] = this.createTokenForm.controls.policies.value;
-    if (policies) {
-      policies.forEach(id =>
-        this.store.dispatch(new AddPolicyMembers(<PolicyMembersMgmtPayload>{
-          id,
-          members: [member]
-        }))
-      );
-    }
+  private assignPolicies(): void {
+    const member = stringToMember(`token:${this.createTokenForm.controls.id.value}`);
```

Comment:
Would this look cleaner if the id was passed? `assignPolicies(tokenID: string, policyIDs: string[]): void`? What do you think?

Reply:
I agree with you: on one hand, it shows a bit more... intent. But on the other hand (which is why I chose to do it this way), I prefer my subscriptions (the call site for this) to be more "high-level" in a sense, typically just dispatching something or setting a property to open/close something, then let the helper functions have the nitty-gritty details.